File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/datasource/DefaultDataSource.java
package com.didichuxing.doraemonkit.kit.performance.datasource;

import com.didichuxing.doraemonkit.kit.performance.widget.LineData;

public class DefaultDataSource implements IDataSource {
    @Override
    public LineData createData() {
        float rate = 50.0f;
        return LineData.obtain(rate, Math.round(rate) + "");
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/manager/datasource/DefaultDataSource.kt
package com.didichuxing.doraemonkit.kit.performance.manager.datasource

import com.didichuxing.doraemonkit.kit.performance.manager.widget.LineData
/**
 *
 * Desc:默认的数据工厂实现类
 * <p>
 * Date: 2020-06-09
 * Updater:
 * Update Time:
 * Update Comments:
 *
 * Author: pengyushan
 */
class DefaultDataSource : IDataSource {
    override fun createData(): LineData? {
        val rate = 50.0f
        return LineData.obtain(rate, Math.round(rate).toString() + "")
    }
}